### ISSUE TITLE
global name 'InputTransformerForm' is not defined

### ISSUE THREAD

**Issue by ion9** [GLab/ToMaTo#951]
An Exception Occured.
Type: NameError
Description: global name 'InputTransformerForm' is not defined

some from main.log
{"category": "auth", "origin": "", "user": "admin", "timestamp": "2015-03-10T02:02:06.440523", "message": "successful login", "caller": [".../tomato/lib/logging.py", 23, "_caller"]}
{"category": "api", "args": [], "user": "admin", "kwargs": {}, "timestamp": "2015-03-10T02:02:06.462367", "caller": [".../tomato/lib/logging.py", 23, "_caller"], "method": "organization_list"}

I do not see anything in the server.log
from a clean install I could not find where the admin user password is stored, so I registered a user, and then edited the user to be like admin, then reset the admin password and when i tryed to add a site i get this error. trying to add a site by had but i'm still reading to work out the attrs to put in to the tomato_sites table. 

this is build on a up to date wheezy openvz running on proxmox

**Comment by ion9** [comment]
Backend version: 3.5.2 / Frontend version: 3.5.2

**Comment by ion9** [comment]
so i now have a site with 

UPDATE tomato_site SET attrs = '{"geolocation":{"longitude":39, "latitude":-83}, "location": "Columbus, Ohio"}' WHERE organization_id = '1';

An Exception Occured.
Type: NoReverseMatch
Description: Reverse for 'tomato.admin.site.edit' with arguments '()' and keyword arguments '{}' not found.

Traceback:
/usr/share/tomato/web/tomato/lib/__init__.py, line 108	in __call__
return self.fun(api, request, *args, **kwargs)
/usr/share/tomato/web/tomato/admin/site.py, line 180	in edit
formargs = [append_empty_choice(organization_name_list(api))]
/usr/share/tomato/web/tomato/admin/__init__.py, line 261	in edit_function
form = Form(*(formargs + clean_formargs), **(clean_formkwargs) )
/usr/share/tomato/web/tomato/admin/site.py, line 139	in __init__
super(EditSiteForm, self).__init__(*args, **kwargs)
/usr/share/tomato/web/tomato/admin/site.py, line 94	in __init__
super(SiteForm, self).__init__(*args, **kwargs)
/usr/share/tomato/web/tomato/admin/__init__.py, line 179	in __init__
super(AddEditForm, self).__init__(data=data, *args, **kwargs)
/usr/share/tomato/web/tomato/admin/__init__.py, line 171	in __init__
super(InputTransformerForm, self).__init__(*args, data=data, **kwargs)
/usr/share/tomato/web/tomato/admin/__init__.py, line 165	in __init__
self.helper.form_action = reverse(self.formaction)
/usr/lib/python2.7/dist-packages/django/core/urlresolvers.py, line 502	in reverse
return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
/usr/lib/python2.7/dist-packages/django/core/urlresolvers.py, line 422	in _reverse_with_prefix
"arguments '%s' not found." % (lookup_view_s, args, kwargs))

**Comment by ion9** [comment]
I'm thinking that this may be a side affect of https://github.com/GLab/ToMaTo/issues/886 but i'm not seeing how to set up the fix. may be see if i can role back the vershion with apt-get?

**Comment by t-gerhard** [comment]
This looks like a problem on the web frontend. You should still be able to run the failing commands via the cli client.
I will examine this later today.

**Comment by t-gerhard** [comment]
@ion9 do these updated files fix these problems?

https://github.com/t-gerhard/ToMaTo/blob/master/web/tomato/admin/site.py
https://github.com/t-gerhard/ToMaTo/blob/master/web/tomato/urls.py

**Comment by ion9** [comment]
i have updated the two files and restarted apache, now i get 
----------------------------------------------------------
An Exception Occured.
Type: TypeError
Description: add_function() takes at least 4 arguments (5 given)

If you continue to encounter this problem, please contact an administrator or file a bug report.
------------------------------------------------------------------
errors from server.log
-------------------------
Shutting down...
Running migrations for tomato:
- Nothing to migrate.
 - Loading initial data for tomato.
Installed 0 object(s) from 0 fixture(s)
Unhandled exception in thread started by <function _stopHelper at 0x33c0848>
Traceback (most recent call last):
  File "/usr/share/tomato/backend/tomato/__init__.py", line 110, in _stopHelper
    if stopped.isSet():
AttributeError: 'NoneType' object has no attribute 'isSet'
Loaded config from /etc/tomato/backend.conf
Loading auth modules...
 -  (internal)
Starting RPC servers
 - :8000, SSL: True
 - :8001, SSL: False
-----------------------------------------

i'm open for any ideas, one other thing i found in the package tomato-backend dose not make a needed directory /var/lib/tomato/templates, had to run  mkdir -p /var/lib/tomato/templates
and in the default web.conf it's set to http, but in backend.conf it calls for ssl
        "SSL_OPTS": {
                "cert_file" : "/etc/tomato/server.cert",
                "key_file": "/etc/tomato/server.cert",
had to edit web.conf and run
  cd /etc/tomato/ && openssl req -new -x509 -days 4000 -nodes -out server.cert -keyout server.cert

**Comment by t-gerhard** [comment]
There is no error on the server side. It simply tells you that there are two ports open, one of which uses SSL, and the other doesn't. This is again a problem with the webfrontend.
I cannot reproduce your problem. Could you tell me what you were trying to do?

**Comment by ion9** [comment]
Currently I'm trying to set up my own tomato backend server, that has a web front on a private VPN, as I do not qualify as a institution to register, nor am I affiliated with one who is registered. 

However I've played with proxmox for many years. And I truly appreciate some of the recent advances with 3.4 for natively supporting ZFS on linux. 
However I've found proxmox to be lacking the ability to easily manage more than one cluster of hosts. 
I believe I can use ToMaTo to solves that requirement, while also enabling my desire for the ability to monitor and profile complex systems in a virtualized environment.

I guess my first question is do we know if individuals can run full backends with web frontends installed from the Debian packages available from the repo?

If not I can take that as a to do item. Document building from get and contribute that to the wiki.
